dot/core/messages.go

```diff
@@ -32,14 +32,15 @@ func (s *Service) validateTransaction(peerID peer.ID, head *types.Header, rt Run
 	externalExt := types.Extrinsic(append([]byte{byte(types.TxnExternal)}, tx...))
 	validity, err = rt.ValidateTransaction(externalExt)
 	if err != nil {
-		if errors.Is(err, runtime.ErrInvalidTransaction) {
+		logger.Debugf("failed to validate transaction: %s", err)
+		switch {
+		case errors.Is(err, runtimererrors.ErrInvalidTxn):
 			s.net.ReportPeer(peerset.ReputationChange{
 				Value:  peerset.BadTransactionValue,
 				Reason: peerset.BadTransactionReason,
 			}, peerID)
+		case errors.Is(err, runtimererrors.ErrUnknownTxn):
 		}
-
-		logger.Debugf("failed to validate transaction: %s", err)
 		return nil, false, nil
```

Comment:
Just double checking, maybe I'm repeating myself; shouldn't we return the unhandled error? Since otherwise it would just silently be discarded. @timwu20 maybe any opinion on that?

Reply:
I do not think so. If we return an error then an error is returned and no peer reporting happens. And submitting an invalid txn is an action that needs to be reported.

Reply:
After further research I think you are right and we should do this.